src/main/java/com/deavensoft/paketomat/courier/CourierServiceImpl.java

```diff
@@ -1,18 +1,38 @@
 package com.deavensoft.paketomat.courier;
 
+import com.deavensoft.paketomat.center.CenterService;
+import com.deavensoft.paketomat.center.model.City;
+import com.deavensoft.paketomat.center.model.Package;
+import com.deavensoft.paketomat.center.model.Status;
+import com.deavensoft.paketomat.city.CityService;
+import com.deavensoft.paketomat.dispatcher.DispatcherService;
+import com.deavensoft.paketomat.exceptions.PaketomatException;
+import lombok.extern.slf4j.Slf4j;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Qualifier;
 import org.springframework.stereotype.Service;
-
+import java.util.ArrayList;
 import java.util.List;
 import java.util.Optional;
 
 @Service
+@Slf4j
 public class CourierServiceImpl implements CourierService {
-    private CourierRepository courierRepository;
+    private final CourierRepository courierRepository;
+    private final CenterService centerService;
+    private final CityService cityService;
+    private final DispatcherService dispatcherService;
+    private List<Package> packagesToDispatch;
+    private List<City> citiesToDispatch;
     @Autowired
-    public CourierServiceImpl(@Qualifier("courier") CourierRepository courierRepository){
+    public CourierServiceImpl(@Qualifier("courier") CourierRepository courierRepository,
```

Comment:
The constructor can be replaced using @ RequiredArgsConstructor at the top of the class.

Reply:
Done